src/features/asciidoctorConfig.ts

```diff
@@ -17,9 +17,28 @@ export async function getAsciidoctorConfigContent (documentUri: vscode.Uri): Pro
   if (workspaceFolder === undefined) {
     return undefined
   }
-  const asciidoctorConfigUri = vscode.Uri.joinPath(workspaceFolder.uri, '.asciidoctorconfig')
+  const configContentFromAsciidoctorConfig = await getConfigContent(workspaceFolder, '.asciidoctorconfig')
+  const configContentFromAsciidoctorConfigDotAdoc = await getConfigContent(workspaceFolder, '.asciidoctorconfig.adoc')
```

Comment:
To simplify the code we can do the following:

1. Create an array of configuration files
2.  Filter `undefined` values in the array
3. If the array is empty, return `undefined`
4. Otherwise, join the array using `\n\n`

Pseudo-code (untested)
```ts
const configContentFromAsciidoctorConfig = await getConfigContent(workspaceFolder, '.asciidoctorconfig')
const configContentFromAsciidoctorConfigDotAdoc = await getConfigContent(workspaceFolder, '.asciidoctorconfig.adoc')

const configContents = [
  configContentFromAsciidoctorConfig,
  configContentFromAsciidoctorConfigDotAdoc
].filter(config => config !== undefined)

if (configContents.length > 0) {
  return configContents.join(`\n\n`)
}
return undefined
```

Reply:
I find it clearer having the 4 cases in different if/else cases than using a loop but the end result is the same and higher chance that you will have to maintain it than me. I updated with your proposal.